### DISCUSSION TITLE
make "stop" command can stop server immediately while server startup.

### DISCUSSION THREAD

**Discussion by kgncengiz** [PaperMC/Paper#10681] [Ideas]
### Is your feature request related to a problem?

sometimes I need to stop the server immediately while startup and to do this I need to kill the process but it is not a good way to shutdown the server shutdown

### Describe the solution you'd like.

make the server listen for the "stop" command while startup and if the "stop" command is executed, the server immediately shuts down.

### Describe alternatives you've considered.

killing the server process in the terminal but it is not a good way to shutdown the server.

**Comment by electronicboy** [comment]
sending stop to a sane console while the process is starting up will allow the command to be read as soon as the console is actively starting, adding extra stop points would just be pointlessly invasive for generally 0 gain, as we'd need to figure out how to manage to tear it back down properly in whatever state you're in, which is just not worth it or viable for a process

**Reply by kgncengiz** [reply]
A optional config option can be added it can be false by default if user make it true than "stop" command immediately kills the Server while sturtup it is not Hard to implement and it is optional so if it breaks anything paper will not be responsible for anything.

**Reply by electronicboy** [reply]
it would require us to have logic in order to hijack the console before the actual console takes over, there is just little gain, and that does not solve the logic of how to shut down properly in any such state, the only known state we know how to shutdown from is when the server is ready to start ticking, and by that point, we already have a fully established terminal which will read any existing command you sent to it from (unless, you're using windows)

**Reply by electronicboy** [reply]
if you just want to shut down promptly from any state with 0 sense, just hit ctrl-c

**Reply by kgncengiz** [reply]
You cant use ctrl+c if you are not using desktop enviroment and using Server in docker or system service with Web ui.

**Reply by lynxplay** [reply]
You can certainly pass SIGINT to a docker container via `docker kill`.
In fact, it is the [default signal](https://docs.docker.com/reference/cli/docker/container/kill/#options) sent by `docker kill`.
